delete_deployment_pods is failing to delete deployment pods

### Description
delete_deployment_pods is failing to delete deployment pods in teardown of some testcases

```
`def finalizer():
        """
        Delete dc pods
        """
        for instance in instances:



      delete_deployment_pods(instance)

tests/conftest.py:1696: 



pod_obj = <ocs_ci.ocs.resources.pod.Pod object at 0x7ff5e58fe7f0>


def delete_deployment_pods(pod_obj):
    """
    Delete a Deployment and all the pods that are controlled by it

    Args:
         pod_obj (Pod): Pod object

    """
    deploy_ocp_obj = ocp.OCP(kind=constants.DEPLOYMENT, namespace=pod_obj.namespace)
    pod_data_list = deploy_ocp_obj.get().get("items")
    if pod_data_list:
        for pod_data in pod_data_list:
            if pod_obj.get_labels().get("name") == pod_data.get("metadata").get("name"):

              deploy_ocp_obj.delete(resource_name=pod_data.get_labels().get("name"))
E                   AttributeError: 'dict' object has no attribute 'get_labels'



ocs_ci/ocs/resources/pod.py:2360: AttributeError`
```

- RP Links : https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/852/31116/1523137/1523186/log
                          https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/852/31116/1523137/1523189/log
- Platform: VSPHERE6-UPI
- Version: ODF - 4.19

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -2357,7 +2357,7 @@ def delete_deployment_pods(pod_obj):
     if pod_data_list:
         for pod_data in pod_data_list:
             if pod_obj.get_labels().get("name") == pod_data.get("metadata").get("name"):
-                deploy_ocp_obj.delete(resource_name=pod_data.get_labels().get("name"))
+                deploy_ocp_obj.delete(resource_name=pod_obj.get_labels().get("name"))
 
 
 def wait_for_new_osd_pods_to_come_up(number_of_osd_pods_before):

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -2358,6 +2358,9 @@ def delete_deployment_pods(pod_obj):
         for pod_data in pod_data_list:
             if pod_obj.get_labels().get("name") == pod_data.get("metadata").get("name"):
                 deploy_ocp_obj.delete(resource_name=pod_obj.get_labels().get("name"))
+                deploy_ocp_obj.wait_for_delete(
+                    resource_name=pod_obj.get_labels().get("name")
+                )
 
 
 def wait_for_new_osd_pods_to_come_up(number_of_osd_pods_before):

